homeassistant/components/madvr/remote.py

```diff
@@ -0,0 +1,74 @@
+"""Support for MadVR remote control."""
+
+from collections.abc import Iterable
+import logging
+from typing import Any
+
+from homeassistant.components.remote import RemoteEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import MadVRConfigEntry
+from .coordinator import MadVRCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MadVRConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the MadVR remote."""
+    coordinator = entry.runtime_data
+    async_add_entities(
+        [
+            MadvrRemote(hass, coordinator, entry.entry_id),
+        ]
+    )
+
+
+class MadvrRemote(CoordinatorEntity[MadVRCoordinator], RemoteEntity):
+    """Remote entity for the MadVR integration."""
+
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: MadVRCoordinator,
+        entry_id: str,
+    ) -> None:
+        """Initialize the remote entity."""
+        super().__init__(coordinator)
+        self.madvr_client = coordinator.client
+        self._attr_name = coordinator.device_info.get("name")
```

Comment:
Integrations should have `_attr_has_entity_name = True`, which prepends the entity name with the device name. To make this entity be the device name, add `_attr_name = None`